### DISCUSSION TITLE
Nesting registryUrl is not possible

### DISCUSSION THREAD

**Discussion by v1ack** [renovatebot/renovate#25759] [Request Help] [answered]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I am trying to get this working for the first time

### Describe the problem

I wanted to use my global preset to describe registry urls, and in some repos add additional registry urls

Custom `defaultRegistryUrls` are ignored if `registryUrls` are set.
Parent `registryUrls` overrides by repo `registryUrls`, because field `registryUrls` is not mentioned as `mergeable: true`

How can I make my own global set registryUrls? Or maybe it's possible to make `registryUrls` mergeable?

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

### Have you created a minimal reproduction repository?

I have explained in the description why a minimal reproduction is impossible

**Comment by rarkins** [comment]
Please describe more generally/higher level what it is you are trying to achieve. Everything you describe is working as designed, and we do not want to make registryUrls mergeable.

**Reply by v1ack** [reply]
In my case there are at least 3 company global registries. And each team can have their own private registry for their own libs, but they also use global registries.
As for now teams can't nest company registries and must mention them in config again

**Reply by v1ack** [reply]
Maybe it's better to add param `parentRegistryUrls`, which will not be overwrited in nested configs

**Reply by rarkins** [reply]
Unfortunately I still don't understand, including what you mean by "nested" registries. This is not a term we use

**Reply by v1ack** [reply]
Company-level config

```json
{
  "packageRules" : [
    {
      "matchDatasources" : ["maven"],
      "registryUrls": [
        "https://repo.company.local/maven-dev",
      ]
    }
  ]
}
```

Expected repo-level config

```json
{
  "extends" : [
    "local>renovate/presets:global-datasources"
  ],
  "packageRules" : [
    {
      "matchDatasources" : ["maven"],
      "registryUrls": [
        "https://repo.company.local/private-team-repo",
      ]
    }
  ]
}
```

As-is repo-level config

```json
{
  "extends" : [
    "local>renovate/presets:global-datasources"
  ],
  "packageRules" : [
    {
      "matchDatasources" : ["maven"],
      "registryUrls": [
        "https://repo.company.local/maven-dev",
        "https://repo.company.local/private-team-repo",
      ]
    }
  ]
}
```

**Reply by rarkins** [reply]
This is working as intended. It's important to be able to override registryUrls

**Reply by v1ack** [reply]
Yes, I understood you

But what about adding additional param as I mentioned before?
`parentRegistryUrls` that won't be overwritten, looks like a great solution. Or something like `allowExtending` in local config

**Reply by rarkins** [reply, marked as answer]
For now, it's not worth us adding complexity if you're the only user who needs it. If we find there are many, we will consider it

**Reply by v1ack** [reply]
Ok, thank you!
